File: Problem Set 2/Vigenere/vigenere.c
/* Program to encrypt a message using the Vigenere cipher */

#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>
#include <stdio.h>

int main(int argc, string argv[])
{
    string keyword = argv[1];
    
    // Check key from command line argument
    while (argc != 2)
    {
        printf("Usage: %s keyword\n", argv[0]);   
        return 1;
    }
    
    // Check command line argument
    for (int i = 0; i < strlen(keyword); i++)
    {
        // Check argument to make sure all characters are alphabetical
        if (!isalpha(keyword[i]))
        {
            printf("Usage: %s keyword\n", argv[0]);   
            return 1; 
        }
    }
    
    // Prompt user for plaintext
    string plaintext = get_string("plaintext: ");

    // Encipher each character in the plaintext string
    for (int i = 0, j = 0; i < strlen(plaintext); i++)
    {
        // Get key for the current letter
        int key = tolower(keyword[j % strlen(keyword)]) - 97;
        
        // If character is an uppercase character
        if (isupper(plaintext[i]))
        {
            plaintext[i] = (((plaintext[i] + key - 65) % 26) + 65);
            j++;
        }
        // If character is a lowercase character
        if (islower(plaintext[i]))
        {
            plaintext[i] = (((plaintext[i] + key - 97) % 26) + 97);
            j++;
        }
        // If character is symbol character
        if (ispunct(plaintext[i]))
        {
            plaintext[i] = plaintext[i];
        }
    }
    
    // Print ciphertext
    printf("ciphertext: %s\n", plaintext);
}

File: Problem Set 6/Vigenere/vigenere.py
# Program to encrypt a message using the Caesar cipher 

import sys 
from cs50 import get_string

def main():
    # Check keyword from command line argument
    while len(sys.argv) != 2:
        sys.exit("Usage: python sys.argv[1] keyword")
            
    # Check argument to make sure all characters are alphabetical
    if sys.argv[1].isalpha() == False:
        sys.exit("Usage: python sys.argv[1] keyword")
        
    # Prompt user for plaintext
    plaintext = get_string("plaintext: ")
        
    # Create a list for ciphertext characters
    ciphertext = []
    key_index = 0
        
    # Encipher each character in the plaintext string
    for c in plaintext:
        if c.isalpha():
            key = ord(sys.argv[1][key_index % len(sys.argv[1])].lower()) - 97
            key_index += 1
            ciphertext.append(cipher(c, key))
        else:
            ciphertext.append(c)
    
    # Print ciphertext
    print("ciphertext: ", end='')
    print("".join(ciphertext))

def cipher(c, key):
    if c.isupper():
        return chr(((ord(c) - 65 + key) % 26) + 65)
    else:
        return chr(((ord(c) - 97 + key) % 26) + 97)
        
if __name__ == "__main__":
    main()
